Support for prlib.ru

andrey-09 opened the pull request:
Добавил поддержку 4.0 качества у картинок с prlib.ru  (картинки через [IIPServer](https://iipimage.sourceforge.io/documentation/protocol). заюзал `JTL` параметр)

Решил, как вы и сказали 3 года назад "[через собирание разных кусочков картинику в одну](https://github.com/aliasn3t/booklead/issues/1#issuecomment-931268629)". Ускорил с помощью asyncio. Занимает по времени где-то 1-2 сек/картинка. 
Код писал, чтоб работал, не в лучших практиках) Если у вас будет время его в вашем хорошем стиле дописать, буду благодарен и поучусь)

Также добавил поддержку https://www.prlib.ru/node/333266/source (до этого только ссылки типа https://www.prlib.ru/item/416819 работали.
Если потребуется, могу побольше закомментить.

Качество картинки (отдаленно и приближенно)
![image](https://github.com/user-attachments/assets/d2b7b43c-d966-48ec-862a-8ca5aff2add4)
![image](https://github.com/user-attachments/assets/051fca85-d9ac-4d1e-83ef-563b7b91b15c)

andrey-09 committed:
Update README.md

andrey-09 committed:
Update README.md

обновил Readme

andrey-09 committed:
Update README.md

andrey-09 committed:
Update README.md

andrey-09 committed:
Bookleads 2.0. Support for prlib.ru. 1-2 seconds/image.

andrey-09 commented:
В будущем были идеи добавить: 
- Оптимизировать PDF? (сейчас средний размер порядка 100 MB
- возможность заранить параллельно

Если вы непротив буду у себя развивать)

andrey-09 committed:
Delete README.md

andrey-09 committed:
Create README.md

andrey-09 commented:
Если вдруг случайно знаете:
я правильно понимаю, что до этого они обрубали клиенту-нам загрузку хорошего качества через параметр [MAX_CVT](https://iipimage.sourceforge.io/documentation/server#:~:text=MAX_CVT%20Limits%20the%20maximum%20image%20dimensions%20in%20pixels) на уровне сервера, поэтому просто через `CVT` плохое качество было?
Или же это просто баг у них такой? Но если баг, то как они на сайт полную картинку выгружают, тоже по частям склеивают?
Или у них какой внутрениий API есть отдельный...

andrey-09 committed:
Update README.md

andrey-09 committed:
Update util.py

Добавил комменты

andrey-09 committed:
Update booklead.py

Закомментил все изменения

andrey-09 committed:
Update booklead.py

andrey-09 committed:
Update booklead.py

andrey-09 committed:
Update booklead.py

Nikolaj-1 commented:
Спасибо за внимание к проекту. Попробовал. Под виндой не пашет.
c:\booklead_2.0-master>python3 booklead.py --list books.txt --pdf y
  File "c:\booklead_2.0-master\booklead.py", line 72
    global results_prlDl
                        ^
IndentationError: unindent does not match any outer indentation level

andrey-09 commented:
@Nikolaj-1 спасибо за ответ, я тож на винде. я кавычку, когда коментил не там поставил.  скачайте пожалуйста заново с [master'a](https://github.com/andrey-09/booklead_2.0).
У меня качает, но возможно ещё что от разных версий чего зависит и т.д. Отпишитесь, как у вас. за сегодня надо бы точно доделать)

andrey-09 committed:
Update requirements.txt

Nikolaj-1 commented:
@andrey-09 всё еще не работает.
c:\booklead_2.0-master>python3 booklead.py --list books.txt --pdf y
c:\booklead_2.0-master\util.py:169: SyntaxWarning: invalid escape sequence '\w'
  s = re.sub("(\w+):", r'"\1":', s)
Ссылок для загрузки: 1
Cсылка: https://www.prlib.ru/item/1286646
Каталог для загрузки: Записки по морской тактике. Ч. 1. Боевые средства флота
Ошибка: unhandled errors in a TaskGroup (1 sub-exception)

Nikolaj-1 commented:
постоянно обрывается,но если каждый раз снова запускать , то докачивает. скачал раза с 10-го. Спасибо!
c:\booklead_2.0-master\util.py:169: SyntaxWarning: invalid escape sequence '\w'
  s = re.sub("(\w+):", r'"\1":', s)
Ссылок для загрузки: 1
Cсылка: https://www.prlib.ru/item/1286646
Каталог для загрузки: Записки по морской тактике. Ч. 1. Боевые средства флота
Ошибка: unhandled errors in a TaskGroup (1 sub-exception)

andrey-09 commented:
@Nikolaj-1 Посмотрю, это asyncio не нравится что-то... 
А у Вас как только начало качать, до конца всю книгу сразу или снова вылетало местами?

andrey-09 commented:
Ок, спасибо, что отписались. как время будет, посмотрю

Nikolaj-1 commented:
постоянно вылетает, чтобы скачать одно книгу надо раз 15-20 запускать, хорошо, что хоть продолжает качать с того места, где оборвалась закачка.

DaNNick22 commented:
Здравствуйте, спасибо огромное.
А можете собрать обновлённый бинарник? 
Мне надо скачать один раз книгу https://www.prlib.ru/item/1287681 в виде изображений в максимальном качестве, без сжатия в pdf, а в Python я не силён

andrey-09 commented:
Здравстуйте, @DaNNick22 прийду домой через час, посмотрю. У меня на компе все качает без ошибок, я вашу книгу себе на закачку поставлю, где-то через 20 мин готовую скину. Так проще будет)  Вам на гугл диск пойдет?  
Бинарники собирать пока не умею)  все равно ещё код надо доработать
Просто к слову, Картинки у вас в папке же находятся, они не удаляются, а pdf можете не делать, просто не подав "`--pdf y`" в терминале.

andrey-09 commented:
@DaNNick22 Гугл диск (картинки в папке, pdf отдельно): https://drive.google.com/drive/folders/1KSzzQhOUyICZGY-wQybNjFPjWSCPLjCa?usp=sharing
Пойдет? Отпишитесь, как скачаете

andrey-09 committed:
Update util.py

solved warning

DaNNick22 commented:
Да, скачал. Спасибо вам огромное!

andrey-09 committed:
Update booklead.py

Добавил предсказатель времени

andrey-09 committed:
Update util.py

Утилиты для времени

andrey-09 committed:
Добавил секунды форматированные и обновил пакеты для Python 3.11.11

andrey-09 committed:
Exceptions of asyncio

andrey-09 committed:
Update README.md

andrey-09 committed:
Update README.md

andrey-09 committed:
Update README.md

andrey-09 committed:
Update README.md

andrey-09 committed:
Update README.md

andrey-09 committed:
Update README.md

andrey-09 committed:
Фиксил проблему с SSL connection. Отладил нетвокринг. Добавил ограничение по времени на request
- добавил парочку новых user agents
- обновил модули (все работает на Python 3.12.10)

andrey-09 committed:
Merge pull request #1 from andrey-09/Local

Фиксил проблему с SSL connection. Отладил нетвокринг. Добавил огранич…

andrey-09 committed:
Delete README.md

andrey-09 committed:
Create README.md

andrey-09 committed:
Merge pull request #2 from andrey-09/To-merge

Networking bug fixes

andrey-09 committed:
Update README.md

andrey-09 commented:
Добрый вечер, @DaNNick22, @Nikolaj-1 . Все проблемы исправил, создал бинарник, при первом запуске секунды 3-4 ничего не выводит (библиотеки подгружает), потом все быстро. Скачать [здесь](https://github.com/andrey-09/booklead_2.0/releases/tag/v2.0-alpha). 
Если проблемы появятся, пишите.

Все вот это пофиксино:

> постоянно вылетает, чтобы скачать одно книгу надо раз 15-20 запускать  

> Каталог для загрузки: Записки по морской тактике. Ч. 1. Боевые средства флота
Ошибка: unhandled errors in a TaskGroup (1 sub-exception)

andrey-09 commented:
@Nikolaj-1 @DaNNick22 Доброго времени суток. пол месяца трудов, зайдите сюда: https://archive.org/details/russian-online-libraries - все книги с библиотеки в свободном доступе, можете зайти и скачать оптимизированный PDF. 
Приятного чтения)

DaNNick22 commented:
> @Nikolaj-1 @DaNNick22 Доброго времени суток. пол месяца трудов, зайдите сюда: https://archive.org/details/russian-online-libraries - почти все книги с библиотеки, можете зайти и скачать оптимизированный PDF (на сайте по страницам можно кликать, и книжку листать) Все можно отфильтровать, поиск даже по тексту внутри книг. Приятного чтения) прошу распространять только знакомым (не хочется, чтоб быстро страйк кинули, если вообще за этим кто смотрит)

Премного благодарен. Вы знаете, я там тоже искал, ещё в прошлом году. А нужна мне книга добавлена лишь 25 мая сего года.
Спасибо огромное!
